### ISSUE TITLE
Detect and fail when more then one AVDL file has same namespace and protocol name

### ISSUE THREAD

**Issue by lwanik** [davidmc24/gradle-avro-plugin#150]
I noticed that since version 0.22.0 where #123 was resolved there are missing records and java classes generated from AVDL files in my projects. It turned out that when two or more AVDL files have the same namespace and protocol name the generated AVPR files contain records from only one of them.
Example:
I have two files:
File1.avdl
```
@namespace("com.company.avro")
protocol Protocol {
  record Record1 {
    string id;
  }
}
```
File2.avdl
```
@namespace("com.company.avro")
protocol Protocol {
  record Record2 {
    string id;
  }
}
```

When I execute `generateAvroJava` task using plugin version 0.21.0 I get File1.avpr with Record1 and File2.avpr with Record2. Generated java classes are Protocol.java, Record1.java and Record2.java.

When I execute `generateAvroJava` task using plugin version 0.22.0 or higher I get only Protocol.avpr with just Record2. Generated java classes are Protocol.java and Record2.java. Record1.java is missing.

It seems that contents of AVPR file are overwritten when you have more than one AVDL file with the same namespace and protocol name (since AVPR file names are based on namespace and protocol name).

**Comment by davidmc24** [comment]
This situation isn't something that this plugin is going to support.  You have two conflicting definitions of a single protocol, `com.company.avro.Protocol`.  That isn't a valid input.  Rename one or both of your protocols, or make them have the same contents (if they're actually redundant copies of the same protocol).

I am open to the possibility of adding support for the plugin to attempt to detect this condition and calling it out as an error rather than silently overwriting.  Would that be of interest to you?

**Comment by lwanik** [comment]
Sure. I think error would be much better than silently overwriting. Currently project fail in code compilation due to missing classes and it is much harder to find the issue. It would be nice to be notified by the plugin about invalid input (duplicated protocols).
Thank you for your answer 🙂

**Comment by davidmc24** [comment]
Fixed via #151
